cmd/snap-update-ns/change.go

```diff
@@ -191,11 +180,9 @@ func (c *Change) ensureTarget(as *Assumptions) ([]*Change, error) {
 			}
 		case "ensure-dir":
 			if !fi.Mode().IsDir() {
-				// Failure to complete because of a file in the way should not be treated as an error.
-				logger.Noticef("WARNING: cannot perform ensure-dir mount for target %q: existing file in the way", path)
+				err = fmt.Errorf("cannot create ensure-dir target %q: existing file in the way", path)
```

Comment:
what's usually the behavior in these situations in other part of this code? this change seems not one we discussed

Reply:
Yes, this change was not part of the last discussion. 
I initially proposed to only issue warnings if `ensire-dir` target is eclipsed by a file, or `must-exist-dir` does not exist.

The reality is, apart form origin `layout` and `overname` and error `IgnoredMissingMount`, [these errors will be issued as a notice anyways](https://github.com/snapcore/snapd/blob/master/cmd/snap-update-ns/update.go#L85).

Now it follows pattern (e.g. [see kind symlink](https://github.com/snapcore/snapd/pull/13342/files#diff-fa7a82a9939947936744df5b4d0d5b74373e4184265ff52a9d56ae22d28a2c9aR179)) to error if the target dir cannot be created because there is a file in the way. This also simplifies mkDirAllWithin to not require specific error definitions anymore.